category/interfaces/graphql/schema.graphql

```diff
@@ -1,3 +1,20 @@
+interface Commerce_Category {
+    code: String!
+    name: String!
+    path: String!
+    active: Boolean!
+    promoted: Boolean!
+}
+
+
+type Commerce_CategoryData implements Commerce_Category {
+    code: String!
+    name: String!
+    path: String!
+    active: Boolean!
+    promoted: Boolean!
+}
+
 interface Commerce_Tree {
     code: String!
     name: String!
```

Comment:
Commerce_CategorySearchResult {
 Category: Commerce_Category
 Product: ProductSearchResult
}

Query 
Commerce_Category(categorycode, Commerce_SearchFilter[]): Commerce_CategorySearchResult

Reply:
Moved logic to category and added filter options